### DISCUSSION TITLE
Tokens connections or Refferencing

### DISCUSSION THREAD

**Discussion by AndresGarciaUXUI** [primefaces/community#1060] [Figma UI Kit]
Hi, could somebody tell me where I can find the status of the token's implementation and what tokens have been referenced? 

I see that some color tokens are referenced, but most of them are not referenced. Are all the tokens going to be referenced to the global colors? See the example below:

Token: calendar.borderColor   Value: {inputList.borderColor}

Are global colors going to be the tokens to referenced to for other colors?

**Comment by onursenture** [comment]
Hello, there is currently no available information about the implementation status of tokens. 

Our UI libraries are continuing with a new Theming API, and alongside this, tokens will transition to their new structure. In this structure, we will implement Primitive / Semantic / Component token sets. This will result in a more user-friendly structure in terms of referencing. 

We cannot share details about the token structure because it's still a work in progress. 

Thanks!

**Reply by heath-capstone** [reply]
Thanks @onursenture for the update!

That said, do you have even a guesstimate on when the Primitive/Semantic/Component token sets might be available? @AndresGarciaUXUI  and I are likely not the only ones looking forward to that update. 😄

**Reply by onursenture** [reply]
Introducing a new token structure to the UI Kit is directly linked with the "CSS Variables" task in the libraries. As such, you can monitor the progression of the "CSS Variables" and "New Figma Tokens" tasks on the following roadmap pages:

https://primevue.org/roadmap/
https://primeng.org/roadmap/
https://primereact.org/roadmap/

**Reply by AndresGarciaUXUI** [reply]
The issue I'm having is that I'm working on creating a branded theme for a project and If I update the theme with my own referenced tokens then next time there is an update then I'll have to re-do my work. It would be nice to know how are the colors are going to be referenced so my work hopefully matches the next update. I'm not sure I'm making sense.

**Reply by onursenture** [reply]
Indeed, we want to avoid a scenario where everything needs redoing. However, considering the level of structural change, there might be no way to prevent this. We're working tirelessly on the API. Once our proof of concept is completed, we'll showcase future changes. As the project's specifics aren't entirely determined yet, any information we provide about color referencing might be misleading. Sometimes, even the work we did last week can become outdated.
